diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 788363a..2226987 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -1,15 +1,15 @@
 # The latest versions are available at https://quiltmc.org/en/usage/latest-versions/
 [versions]
-minecraft = "1.20.4"
-quilt_mappings = "1.20.4+build.3"
+minecraft = "1.21"
+quilt_mappings = "1.21+build.2"
 
 quilt_loom = "1.7.3"
 quilt_loader = "0.26.0"
 
-fabric_api = "0.97.1+1.20.4"
+fabric_api = "0.100.4+1.21"
 
-modmenu = "9.0.0"
-ears = "1.4.6+fabric-1.20.2"
+modmenu = "11.0.1"
+ears = "1.4.6"
 
 [libraries]
 minecraft = { module = "com.mojang:minecraft", version.ref = "minecraft" }
diff --git a/src/main/java/net/orifu/skin_overrides/screen/AlwaysSelectedEntryGridWidget.java b/src/main/java/net/orifu/skin_overrides/screen/AlwaysSelectedEntryGridWidget.java
index e08cc5c..a79b3c4 100644
--- a/src/main/java/net/orifu/skin_overrides/screen/AlwaysSelectedEntryGridWidget.java
+++ b/src/main/java/net/orifu/skin_overrides/screen/AlwaysSelectedEntryGridWidget.java
@@ -39,7 +39,7 @@ protected int getRowTop(int index) {
     }
 
     @Override
-    protected void renderList(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+    protected void method_25311(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
         int baseX = this.getRowLeft();
         int w = this.itemWidth;
         int h = this.itemHeight - 4;
diff --git a/src/main/java/net/orifu/skin_overrides/screen/LibraryScreen.java b/src/main/java/net/orifu/skin_overrides/screen/LibraryScreen.java
index 87f344e..44eab8a 100644
--- a/src/main/java/net/orifu/skin_overrides/screen/LibraryScreen.java
+++ b/src/main/java/net/orifu/skin_overrides/screen/LibraryScreen.java
@@ -284,7 +284,7 @@ public void filesDragged(List<Path> paths) {
     private void addFromSearch() {
         var maybeProfile = ProfileHelper.idToProfile(this.searchBox.getText());
         if (maybeProfile.isEmpty()) {
-            this.client.getToastManager().add(new SystemToast(SystemToast.C_ozahoshp.field_47585,
+            this.client.getToastManager().add(new SystemToast(SystemToast.Id.PACK_LOAD_FAILURE,
                     Text.translatable("skin_overrides.no_profile.title", this.searchBox.getText()),
                     Text.translatable("skin_overrides.no_profile.description")));
         } else {
diff --git a/src/main/java/net/orifu/skin_overrides/util/ProfileHelper.java b/src/main/java/net/orifu/skin_overrides/util/ProfileHelper.java
index 4354968..bd42bdc 100644
--- a/src/main/java/net/orifu/skin_overrides/util/ProfileHelper.java
+++ b/src/main/java/net/orifu/skin_overrides/util/ProfileHelper.java
@@ -34,7 +34,7 @@ public static GameProfile idToBasicProfile(String id) {
             profile = getUserCache().findByName(id);
         }
 
-        return profile.orElseGet(() -> UuidUtil.method_54140(id));
+        return profile.orElseGet(() -> UuidUtil.createOfflinePlayerProfile(id));
     }
 
     public static Optional<GameProfile> idToProfile(String id) {
